descartes_planner/src/sparse_planner.cpp

```diff
@@ -526,23 +526,25 @@ bool SparsePlanner::getPath(std::vector<TrajectoryPtPtr>& path)
   for(int i = 0; i < cart_points_.size();i++)
   {
     TrajectoryPtPtr p = cart_points_[i];
-    JointTrajectoryPt& j = joint_points_map_[p->getID()];
+    const JointTrajectoryPt& j = joint_points_map_.at(p->getID());
     path[i] = TrajectoryPtPtr(new JointTrajectoryPt(j));
   }
 
   return true;
 }
 
-int SparsePlanner::getErrorCode()
+int SparsePlanner::getErrorCode() const
 {
   return error_code_;
 }
 
-bool SparsePlanner::getErrorMessage(int error_code, std::string& msg)
+bool SparsePlanner::getErrorMessage(int error_code, std::string& msg) const
 {
-  if(error_map_.count(error_code)>0)
```

Comment:
Just out of curiosity, was there anything incorrect about the previous approach?